src/main.rs

```diff
@@ -1,9 +1,13 @@
+use crate::usage::usage;
```

Comment:
I think this could be simplified to
```rust
use usage::usage;
```

Reply:
I would rather leave it in.  The plan is to also outsource the types you introduced to dedicated modules.  When importing them, the `crate ::` part will indicate to human readers that these modules belong to this project.

Reply:
That would be a good candidate for creating a separate module. But I think the signal, that there are files by that name in the same directory should be signal enough to human readers. 😅

Reply:
Done.

By the way, I sorted the import lines by alphabet, this is, moving the `use usage::usage;` under the import of the `std` crates.  In my opinion, that way, it is easier to keep track of which modules are imported.

Reply:
All right. 😄 

You might check if `rust fmt` changes the order. I think the `rust-analyser` extension for vs code does in its default settings.

Reply:
Would it be an idea to add something like a pre-commit check or GitHub Action to this repository to check for such things and correct them automatically, at option?

Reply:
Sure, I just never worked with these things. 😅